projects/kit/components/calendar-range/calendar-range.component.ts

```diff
@@ -186,6 +182,9 @@ export class TuiCalendarRange implements OnInit, OnChanges {
             this.selectedActivePeriod = item;
             this.updateValue(item.range.dayLimit(this.min, this.max));
             this.itemChange.emit(item);
+            this.defaultViewedMonth =
+                (this.items.length ? this.value?.to : this.value?.from) ||
+                this.defaultViewedMonth;
```

Comment:
duplicate 
https://github.com/taiga-family/taiga-ui/pull/8843/files#diff-8fda4881ec9c2cd09a2bf32685bee1e50c9f4880e0fb41586f6e88bc0001e3e7R117-R118

could you take it out to the separate method?

Reply:
Oh yeah .  i think we don't need to check the items here , Let me fix it .

Reply:
@mohan-mu the same thing can be carried out in the general method :)

https://github.com/taiga-family/taiga-ui/pull/8843/files#diff-8fda4881ec9c2cd09a2bf32685bee1e50c9f4880e0fb41586f6e88bc0001e3e7R99

and 

https://github.com/taiga-family/taiga-ui/pull/8843/files#diff-8fda4881ec9c2cd09a2bf32685bee1e50c9f4880e0fb41586f6e88bc0001e3e7L122